core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1799,6 +1799,20 @@ class SparkContext(config: SparkConf) extends Logging {
         // For local paths with backslashes on Windows, URI throws an exception
         addJarFile(new File(path))
       } else {
+        /**
+          * Check Path valid
+          */
+        val uriPath = new Path(path).toUri
+        val schemeCorrectedPath = uriPath.getScheme match {
+          case null => new File(path).getCanonicalFile.toURI.toString
+          case "local" => "file:" + uriPath.getPath
+          case _ => path
+        }
+        val hadoopPath = new Path(schemeCorrectedPath)
+        val fs = hadoopPath.getFileSystem(hadoopConfiguration)
+        if(!fs.exists(hadoopPath))
+          throw new FileNotFoundException(s"Jar ${schemeCorrectedPath} not found")
```

Comment:
If anything, why not check this below?

Reply:
When we use "ADD JAR" SQL command, it will call SessionResourceBuilder's addJar method.Then it call SparkContext's addJar method. It truly happen that when we add jar path with HDFS schema, it don't check . May be we can add this check in SessionResourceBuilder?

Reply:
> If anything, why not check this below?
When we use "ADD JAR" SQL command, it will call SessionResourceBuilder's addJar method.Then it call SparkContext's addJar method. It truly happen that when we add jar path with HDFS schema, it don't check . May be we can add this check in SessionResourceBuilder?

Reply:
@srowen  I change this check to SessionResourceBuilder.   Then only sql query will cause this check, won't impact start process

Reply:
What is the potential impact if we add this change in `SparkContext#addJar`? What I can think of is that will delay the start process as each remote jar will be checked. Also do we need to add a similar check in `SparkContext#addFile` API?

Reply:
@jerryshao when Add File, it will call fs.getFileStatus,  it will check if the path is a file or a dir, this action will return exception when we add a wrong path of file.  

`19/06/20 14:59:45 ERROR org.apache.spark.sql.hive.thriftserver.SparkExecuteStatementOperation: "Error executing query, currentState RUNNING, "
java.io.FileNotFoundException: /userd
	at org.apache.hadoop.fs.viewfs.InodeTree.resolve(InodeTree.java:403)
	at org.apache.hadoop.fs.viewfs.ViewFileSystem.getFileStatus(ViewFileSystem.java:377)
	at org.apache.spark.SparkContext.addFile(SparkContext.scala:1546)
	at org.apache.spark.SparkContext.addFile(SparkContext.scala:1510)
	at org.apache.spark.sql.execution.command.AddFileCommand.run(resources.scala:50)
	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult$lzycompute(commands.scala:70)
	at org.apache.spark.sql.execution.command.ExecutedCommandExec.sideEffectResult(commands.scala:68)
	at org.apache.spark.sql.execution.command.ExecutedCommandExec.executeCollect(commands.scala:79)
	at org.apache.spark.sql.Dataset$$anonfun$6.apply(Dataset.scala:195)
	at org.apache.spark.sql.Dataset$$anonfun$6.apply(Dataset.scala:195)
	at org.apache.spark.sql.Dataset$$anonfun$53.apply(Dataset.scala:3365)
	at org.apache.spark.sql.execution.SQLExecution$.withCustomJobTag(SQLExecution.scala:119)
	at org.apache.spark.sql.execution.SQLExecution$$anonfun$withNewExecutionId$1.apply(SQLExecution.scala:79)
	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:143)
	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:73)
	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:3364)
	at org.apache.spark.sql.Dataset.<init>(Dataset.scala:195)
	at org.apache.spark.sql.Dataset$.ofRows(Dataset.scala:80)
	at org.apache.spark.sql.SparkSession.sql(SparkSessi`

Reply:
@AngersZhuuuu would you please answer this question? I saw that most of your codes are just copied from `SparkContext#addJar`, I'm wondering why do we make this change at there?

> What is the potential impact if we add this change in SparkContext#addJar?

Reply:
@jerryshao  First I put this code at SparkContext#addJar, but for srowen's comment, I change it to SessionState, Since this problem happen when we write wrong "ADD JAR " SQL . The code is nothing special so I just modify some code from addJar & addFile then move to here

Reply:
I think this problem does not only exist in using `ADD JAR`, normally if you call `SparkContext#addJar`, it will also be failed. So my think is that it could be fixed in `addJar`, rather than a separate method.

Reply:
@jerryshao  I was to focused on SQL engine. you said is right.  Maybe I should check more with @srowen.

Reply:
@jerryshao How about my latest change .

Reply:
To me I would prefer to add the check in `addJar` not a separate method, which also keep align with `addFile` (it will also throw an exception in place when file is not found).

Reply:
@jerryshao  sorry, when I @ you, I forget to push mu code from local to GitHub. ==